Blog post introducing OSF (#717)
ropensci/roweb2: content/blog/2020-08-04-osf.md

---
title: Getting to Know OSF
author:
  - Aaron Wolen
date: '2020-08-04'
slug: osf-and-r
categories:
  - blog
tags:

>   - open science

Reviewer: ```suggestion
  - open-science
  - osf
  - osfr
```

  - reproducibility

>   - datasharing

Reviewer: ```suggestion
  - reproducible-research
  - datasharing
  - data-access
```

package_version: 0.1.0
description: >
  An introduction to the Open Science Framework (OSF).

> twitterImg: blog/2019/06/04/post-template/name-of-image.png

Reviewer: ```suggestion

```

Before diving in, I do want to acknowledge that OSF is pretty multi-faceted, and includes a number of other services outside of its core project management functionality, including things like [pre-print servers][osf-preprint] and a [research registration repository][osf-registry]. While incredibly cool and noteworthy, those aspects of OSF fall outside the scope of this post, which I'm limiting to project management features.

### Is OSF like a GitHub for research?

It's not a perfect analogy as the two services differ in important ways that I'll discuss below, but if you're already familiar with GitHub it certainly provides a useful mental model to get a sense of what OSF offers. For example, like GitHub, OSF is a hosted project management service organized around project repositories that can be private, shared with select collaborators, or made publicly accessible. OSF projects also include cloud storage for uploading project files, automatic file versioning, markdown-based wikis for documentation and/or shared task lists, traffic analytics to assess impact and engagement, and they can even be "forked" by other users.

### How are they different?

> Ultimately the two services are different because they are designed for different purposes: facilitating software development vs facilitating scientific research. While the line separating these tasks is often blurry (or non-existant), supporting research requires accommodating files of all shapes and sizes, whereas software (generally) comes in one flavor: plain text. Furthermore, GitHub is intended for software developers, an inherently technical userbase, whereas OSF is designed to accomodate users with a much wider range of computational expertise.

Reviewer: non-existent
user base
accommodate
```suggestion
Ultimately the two services are different because they are designed for different purposes: facilitating software development vs facilitating scientific research. While the line separating these tasks is often blurry (or non-existent), supporting research requires accommodating files of all shapes and sizes, whereas software (generally) comes in one flavor: plain text. Furthermore, GitHub is intended for software developers, an inherently technical user base, whereas OSF is designed to accommodate users with a much wider range of computational expertise.
```

One area where the manifestation of these differences can be seen is in how OSF handles file versions. While OSF automatically stores all previous versions of files as new ones are uploaded, it's not a true version control system in the sense of something like git, which intelligently stores only the differences between file versions, making it possible to easily compare versions and automatically detect conflicting changes. On OSF, the latest version of a file is simply the last uploaded version regardless of how or whether its content changed. However, what OSF's versioning feature lacks in power and options, it makes up for in accessibility---another key differentiator. Creating a new version on OSF simply requires dragging and dropping an updated file onto the project's file browser

Version control is one feature common to both GitHub and OSF, but let's look at some of the key features that are unique OSF that are purpose built for managing and disseminating research.

#### Integrations

> First and foremost, I want to point out that adopting (or just testing out) OSF does not require upending your current workflows and abandoning services you're already happy with. That's because OSF directly integrates with many popular services through its [add-ons system][osf-addons]. So although OSF provides its own free cloud storage, if you're already using Dropbox, Google Drive, S3, etc, then you can simply add the relevent folder/bucket from that service to your OSF project. And if you're using GitHub/GitLab to manage your source code (good idea), then you can also link the relevant repository. Either way, they all show-up as first-class citizens in your OSF project, right alongside OSF's native cloud storage buckets.

Reviewer: relevant
```suggestion
First and foremost, I want to point out that adopting (or just testing out) OSF does not require upending your current workflows and abandoning services you're already happy with. That's because OSF directly integrates with many popular services through its [add-ons system][osf-addons]. So although OSF provides its own free cloud storage, if you're already using Dropbox, Google Drive, S3, etc, then you can simply add the relevant folder/bucket from that service to your OSF project. And if you're using GitHub/GitLab to manage your source code (good idea), then you can also link the relevant repository. Either way, they all show-up as first-class citizens in your OSF project, right alongside OSF's native cloud storage buckets.
```

#### Storage

> OSF provides a generous amount of cloud storage for projects, limiting only the size of individual files to < 5Gb. This allows you to store all of your project materials on OSF, even large data files. Furthermore, COS takes [data preservation seriously][osf-preservation], guaranteeing that, in the unlikely event that COS ceases to exist, your files will remain accessible for at least another 50 years from the day it closes its doors.

Reviewer: What is COS? Consider defining this acronym here

Author: Ah, thanks! I cut the paragraph it was originally defined in and forgot to re-define.
